modules/auxiliary/scanner/http/dolibarr_16_contact_dump.rb

```diff
@@ -0,0 +1,151 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Auxiliary::Scanner
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Auxiliary::Report
+  include Msf::Module::Failure
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Dolibarr 16 pre-auth contact database dump',
+        'Description' => %q{
+          Dolibarr version 16 < 16.0.5 is vulnerable to a pre-authentication contact database dump.
+          An unauthenticated attacker may retreive a company’s entire customer file, prospects, suppliers,
+          and potentially employee information if a contact file exists.
+          Both public and private notes are also included in the dump.
+        },
+        'Author' => [
+          'Vladimir TOUTAIN', 'Nolan LOSSIGNOL-DRILLIEN'
+        ],
+        'License' => MSF_LICENSE,
+        'DisclosureDate' => '2023-03-14',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [],
+          'SideEffects' => [IOC_IN_LOGS]
+        },
+        'References' => [
+          ['URL', 'https://www.dsecbypass.com/en/dolibarr-pre-auth-contact-database-dump/'],
+          ['URL', 'https://github.com/Dolibarr/dolibarr/blob/16.0.5/ChangeLog#L34'],
+          ['URL', 'https://github.com/Dolibarr/dolibarr/commit/bb7b69ef43673ed403436eac05e0bc31d5033ff7'],
+          ['URL', 'https://github.com/Dolibarr/dolibarr/commit/be82f51f68d738cce205f4ce5b469ef42ed82d9e']
+        ]
+      )
+    )
+    register_options(
+      [
+        Opt::RPORT(80),
+        OptString.new('TARGETURI', [true, 'Path to Dolibarr instance', '/'])
+
+      ]
+    )
+  end
+
+  def check_host(_ip)
```

Comment:
Returning an array of check code and software version (which may or may not be nil) seems kind of messy.

I realise that this is due to using the `check_host` method from within `exploit` method. This becomes even more messy in the `exploit` method where you throw out both values and YOLO exploitation anyway.

Personally I would create a method to retrieve the version and a method to check whether a given version is vulnerable.